staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -84,9 +88,22 @@ type RequestDigest struct {
 // this type and cfgMeal follow the convention that the suffix
 // "Locked" means that the caller must hold the configController lock.
 type configController struct {
+	name             string // varies in tests of fighting controllers
+	clock            clock.PassiveClock
 	queueSetFactory  fq.QueueSetFactory
 	obsPairGenerator metrics.TimedObserverPairGenerator
 
+	// Puts something in the workqueue to trigger work
+	finishHandlingNotification func(workqueue.RateLimitingInterface, interface{})
+
+	// How this controller appears in an ObjectMeta ManagedFieldsEntry.Manager
+	asFieldManager string
+
+	// Given a boolean indicating whether a FlowSchema's referenced
+	// PriorityLevelConfig exists, return a boolean indicating whether
+	// the reference is dangling
+	foundToDangling func(bool) bool
```

Comment:
IIUC, this is to allow testing of the functionality.
Instead of making this a parameter, can we actually unit-test at lower level of functions?

Reply:
I am trying to make a behavioral test of high-level units rather than a checksum test of low-level functions.

Reply:
But for behavioral test, why do you need to overwrite this?

Reply:
Expanded explanation in TestableConfig.